File: src/main/resources/public/js/piece-factory.js
const PIECES = {
    BLACK_KING: `<img src="../img/king_black.png" class="piece" alt="king_black"/>`,
    BLACK_QUEEN: `<img src="../img/queen_black.png" class="piece" alt="queen_black"/>`,
    BLACK_ROOK: `<img src="../img/rook_black.png" class="piece" alt="rook black"/>`,
    BLACK_BISHOP: `<img src="../img/bishop_black.png" class="piece" alt="bishop black"/>`,
    BLACK_KNIGHT: `<img src="../img/knight_black.png" class="piece" alt="knight black"/>`,
    BLACK_PAWN: `<img src="../img/pawn_black.png" class="piece" alt="pawn black"/>`,
    WHITE_KING: `<img src="../img/king_white.png" class="piece" alt="king white"/>`,
    WHITE_QUEEN: `<img src="../img/queen_white.png" class="piece" alt="queen white"/>`,
    WHITE_ROOK: `<img src="../img/rook_white.png" class="piece" alt="rook_white"/>`,
    WHITE_BISHOP: `<img src="../img/bishop_white.png" class="piece" alt="bishop white"/>`,
    WHITE_KNIGHT: `<img src="../img/knight_white.png" class="piece" alt="knight white"/>`,
    WHITE_PAWN: `<img src="../img/pawn_white.png" class="piece" alt="pawn white"/>`,
}

document.addEventListener("DOMContentLoaded", buildPieces);

function buildPieces() {
    fetch("/chess")
        .then(res => res.json())
        .then(data => {
            if (data.body.turn === "BLACK") {
                document.getElementById("black-versus").classList.add("currentTurn");
                document.getElementById("white-versus").classList.remove("currentTurn");
            } else {
                document.getElementById("black-versus").classList.remove("currentTurn");
                document.getElementById("white-versus").classList.add("currentTurn");
            }
            Array.from(data.body.boardDTO.pieceDTOS)
                .filter(piece => piece.name !== "BLANK")
                .forEach(piece => {
                    const position = piece.position;
                    const pieceName = piece.color + "_" + piece.name;
                    document.getElementById(position).innerHTML = PIECES[pieceName];
                })
        })
}

File: src/main/java/chess/utils/PieceFactory.java
package chess.utils;

import chess.domain.piece.Bishop;
import chess.domain.piece.Color;
import chess.domain.piece.EmptyPiece;
import chess.domain.piece.King;
import chess.domain.piece.Knight;
import chess.domain.piece.Pawn;
import chess.domain.piece.Piece;
import chess.domain.piece.Queen;
import chess.domain.piece.Rook;
import java.util.HashMap;
import java.util.Map;

public class PieceFactory {

    private static final Map<String, Piece> factory = new HashMap<>();

    static {
        factory.put("white_pawn", new Pawn(Color.WHITE));
        factory.put("white_knight", new Knight(Color.WHITE));
        factory.put("white_bishop", new Bishop(Color.WHITE));
        factory.put("white_rook", new Rook(Color.WHITE));
        factory.put("white_queen", new Queen(Color.WHITE));
        factory.put("white_king", new King(Color.WHITE));

        factory.put("black_pawn", new Pawn(Color.BLACK));
        factory.put("black_knight", new Knight(Color.BLACK));
        factory.put("black_bishop", new Bishop(Color.BLACK));
        factory.put("black_rook", new Rook(Color.BLACK));
        factory.put("black_queen", new Queen(Color.BLACK));
        factory.put("black_king", new King(Color.BLACK));

        factory.put("empty", EmptyPiece.getInstance());
    }

    private PieceFactory() {
    }

    public static Piece convertToPiece(String pieceName) {
        if (!factory.containsKey(pieceName)) {
            throw new IllegalStateException("잘못된 pieceName이 들어왔습니다.");
        }
        return factory.get(pieceName);
    }
}
